pkg/parse/parse.go

```diff
@@ -414,6 +428,130 @@ func GetBindMount(args []string, contextDir string) (specs.Mount, error) {
 	return newMount, nil
 }
 
+// GetCacheMount parses a single cache mount entry from the --mount flag.
+func GetCacheMount(args []string) (specs.Mount, error) {
+	var err error
+	newMount := specs.Mount{
+		Type: TypeBind,
+	}
+
+	setDest := false
+	bindNonRecursive := false
+	setShared := false
+	setReadOnly := false
+	// if id is set a new subdirectory with `id` will be created under /host-temp/buildah-build-cache/id
+	id := ""
+	//buidkit parity: cache directory defaults to 755
+	mode := 0755
+	//buidkit parity: cache directory defaults to uid 0 if not specified
+	uid := 0
+	//buidkit parity: cache directory defaults to gid 0 if not specified
+	gid := 0
+
+	for _, val := range args {
```

Comment:
This whole section seems to be a cut and paste, can we create a shared piece of code for handling options?